### ISSUE TITLE
incorrect date and time values at line, additional questions

### ISSUE THREAD

**Issue by tomasvanek** [Tim55667757/PriceGenerator#24]
Hi, I am trying to generate data similar to SP500 or DJ via this command with history more than 1 year and I am receiving these issues:

commnad:
pricegenerator --timeframe 1 --start "2019-01-02 00:00" --precision 1  --trend-deviation 0.5 --save-to sp3.csv --max-close 8500 --min-close 2000 --generate --horizon 5000000 --debug-level 10 --max-body 150 --up-candles-prob 0.80 --split-trend up  --split-count 100

- time series ending everytime at line 439261 because of invalid date and time there should be 2:00 not 1:00 as you can see on this screen. so the date and time does not continue in right way.
![image](https://github.com/Tim55667757/PriceGenerator/assets/15350149/58136cf7-cff6-4d86-9d06-2291985e17cd)

- is there a way on how to generate data from existing data with correlation lets say 0.7? input should be csv file 
- is there an existing way to generate data similar to existing ticker like SP500 ? I am receivng data as you can see on the screen without long trends 
![image](https://github.com/Tim55667757/PriceGenerator/assets/15350149/81d98b1f-6435-45d3-bb1a-58c955b73308)
- in the history it is important to have data without gaps or minus price values, in my testing I was receiving some samples with low negative vlaues. How can I avoid to these situation? 
- I would like try generate data lets say for 10 years, is there a way? If you can help me with these tasks I can pay you for a custom coding.

Thank you for your answer.

**Comment by Tim55667757** [comment]
Hello, @tomasvanek !

I'll think about your questions this week and try to come up with something.

**Comment by tomasvanek** [comment]
Thank you.

**Comment by tomasvanek** [comment]
Hi @Tim55667757 please, did you have time to take a look at it? Thanks

**Comment by Tim55667757** [comment]
Hi, @tomasvanek !

1) Yes, I tried to reproduce the first error until it worked. Perhaps this has something to do with the local startup time. I have GMT+3, and the series is generated normally. I'll try to fix this problem.

2) I don’t really understand what exactly needs to be done. Can you create a new issue and describe how you see it? At now each candle is generated separately from full series. The entire series is unknown in advance. It’s not clear how you can make a candle that is 70% similar to another candle? In relation to candles, there is no such measure of “similarity”. It is also not clear how to bring an already generated series to the required correlation?

3) Yes, it's possible. For long periods of up trend, down trend or no trend, the `--split-trend` and `--split-count` parameters are used. How to set trends, see more details here: https://github.com/Tim55667757/PriceGenerator#Split-data-by-some-trends
For example, if you have 5 million candles, here's how to define three long periods: rising (1 million), falling (2 million), then no trend (2 million):
`--split-trend=up-down-no --split-count 1000000 2000000 2000000`
In all cases, it is better to always make the `--up-candles-prob` parameter close to `0.5`, e. g. `0.48`-`0.52`.

4) Negative numbers really show up in long series. Thanks for noticing! I'll fix this ASAP.

5) To generate data for 10 years or another period, you can calculate how much it will be approximately in minutes and specify it in the `--horizon` parameter. For example, `10 * 365 * 24 * 60 = 5 256 000` minutes.

**Comment by Tim55667757** [comment]
Hi, @tomasvanek !

Sorry for the long wait, I tried to test various options for the situation described in Task 1, but was not able to generate a similar data chain. Also, I did not find a similar case with the Pandas library on StackOverflow. You can see for yourself, here is a very simple code that generates a sequence of dates in a standard way. I'm not changing the default Pandas settings in any way here. Can you give more details on the first error? How to reproduce it?

It seems I have fixed most of another errors here in #24 and in #25. Negative numbers should no longer appear. The quality of generated candle tail outliers and candle volumes has also been improved.

**You can use and check up developers build v1.4.dev90 for your needs. Please, use hard lock the version:  `pip install pricegenerator==1.4.dev90`, until stable release 1.4 comes out.** PyPI: https://pypi.org/project/pricegenerator/1.4.dev90/

To generate a lot of candles, also with up and down trends, please, test this command locally:
```
pricegenerator --debug-level 10  --generate --save-to test.csv --render-bokeh test.html --timeframe 1 --start "2019-01-02 00:00" --precision 1 --trend-deviation 1.5 --max-close 8500 --min-close 2000 --max-volume 5000000 --max-body 150 --horizon 5000 --up-candles-prob 0.51 --split-trend up-down-no-up --split-count 1000 2000 500 1500 --outliers-prob 0.03
```
With a similar command I got this result:
![image](https://github.com/Tim55667757/PriceGenerator/assets/7660870/95c6ba92-07d1-4384-b040-29b2c4fbe50d)


Warning! Calculations and saving for a lot of candles takes more time. I would suggest breaking the data into chunks, no more than a year in length (~500K candles).

Please check it out and let me know if I helped you.

**Comment by Tim55667757** [comment]
I hope the latest release build will be useful for your: https://pypi.org/project/pricegenerator/1.4.93/

Release-notes: https://github.com/Tim55667757/PriceGenerator/releases/tag/v1.4.93
